test/acceptance/features/bindAppToService.feature

```diff
@@ -585,3 +585,103 @@ Feature: Bind an application to a service
+                services:
+                -   group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: backend-demo-2
+            """
+        Then jq ".status.conditions[] | select(.type=="CollectionReady").status" of Service Binding "binding-request-backend-ann-sb" should be changed to "True"
+        And jq ".status.conditions[] | select(.type=="InjectionReady").status" of Service Binding "binding-request-backend-ann-sb" should be changed to "False"
+        And Secret "binding-request-backend-ann-sb" contains "BACKEND_HOST" key with value "example.com"
+        And Secret "binding-request-backend-ann-sb" contains "BACKEND_READY" key with value ""
+
+        # Backend metadata.annotations for service binding is updated
+        When The Custom Resource is present
+            """
+            apiVersion: "stable.example.com/v1"
+            kind: Backend
+            metadata:
+                name: backend-demo-2
+                annotations:
+                    servicebindingoperator.redhat.io/status.ready: 'binding:env:attribute'
+                    servicebindingoperator.redhat.io/spec.host: 'binding:env:attribute'
+            spec:
+                host: example.com
+            status:
+                ready: true
+            """
+        Then Secret "binding-request-backend-ann-sb" contains "BACKEND_READY" key with value "true"
+        And Secret "binding-request-backend-ann-sb" contains "BACKEND_HOST" key with value "example.com"
+
+   @negative
+   Scenario: Backend Service metadata annotations update not specific to service bindings does not get propagated to the binding secret
+        Given OLM Operator "backend" is running
+        * The Custom Resource is present
+            """
+            apiVersion: "stable.example.com/v1"
+            kind: Backend
+            metadata:
+                name: backend-demo-3
+                annotations:
+                    host : 'binding:env:attribute'
+            spec:
+                host: example.com
+            status:
+                ready:true
+            """
+        * Service Binding is applied
+            """
+            apiVersion: operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: binding-request-backend-ann
+            spec:
+                services:
+                -   group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: backend-demo-3
+            """
+        Then jq ".status.conditions[] | select(.type=="CollectionReady").status" of Service Binding "binding-request-backend-ann" should be changed to "True"
+        And jq ".status.conditions[] | select(.type=="InjectionReady").status" of Service Binding "binding-request-backend-ann" should be changed to "False"
+        And Secret "binding-request-backend-ann" contains "BACKEND_HOST" key with value ""
```

Comment:
why should secret contain at all `BACKEND_HOST` key? If I understand correctly, that should not be the case.

Reply:
"BACKEND_HOST" key with value "" means that the key also does not exist. If the key exists and the value does not exist then the value is `<no-value>` whereas when both key and value are not present then for whatever key we put we get a value as `""` So this step means that there is no key and value .

Reply:
IMHO, this interpretation is confusing, especially for users. If a key does not exist in the secret, we should state that in the appropriate step. For example:

```
Secret XXX does not contain key KEY
```

Please implement such a step.

Reply:
updated